src/libraries/Microsoft.PowerFx.Core/Public/Values/UntypedObjectValue.cs

```diff
@@ -43,7 +43,12 @@ public interface IUntypedObject
         // It is validated to be in a culture invariant standard format number, possibly with dot decimal separator and "E" exponent.
         // This method need not be implemetned if ExteranlType.UntypedNumber is not used.
         // GetDouble and GetDecimal can be called on an ExternalType.UntypedNumber.
-        string GetUntypedNumber();
+        string GetUntypedNumber();
+
+        /// <summary>
+        /// Returns the list of available properties if the current instance is an object.
+        /// </summary>
+        IEnumerable<string> GetProperties();
```

Comment:
Some UO implementations may not be able to enumerate their properties - such as if they're lazy or exist over an open system.  Can you make this:

`bool TryGetProperties(out IEnumerable<string> names)`

This also avoids the NotImpl exceptions and lets us just return false.